app/_src/gateway/plugin-development/wasm.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Create a Proxy-Wasm filter
+book: plugin_dev
+chapter: 11
+---
+
+## Your first Proxy-Wasm filter
+
+Filters can be written as a compilation target in almost any language that supports WebAssembly, but the easiest way to get started is by
+using a language which has an existing [Proxy-Wasm](https://github.com/proxy-wasm/spec)
+SDK, such as [Go](https://github.com/tetratelabs/proxy-wasm-go-sdk/)
+or [Rust](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/).
+
+## Coding, building and deploying a Proxy-Wasm filter
+
+Kong has made Go and Rust Proxy-Wasm filter templates available. They can be used
+to start writing your own filters, or to build as they are and get your first `hello
+world`-type filter:
+
+* [Go Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-go-filter-template/)
+* [Rust Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-rust-filter-template/)
+
+### Deploying the filter with Kong
+
+After writing and building the filter, the next steps are to configure Kong to enable Wasm support and add the filter, then associate the filter to a route or a service. You will then be able to start Kong and issue requests that go through the filter.
+
+#### Configuration
+
+WebAssembly support must be enabled in {{site.base_gateway}}:
+
+```console
+$ export KONG_WASM=on
+```
+
+Additionally, the `wasm_filters_path` parameter must be configured in order for
+{{site.base_gateway}} to load the filter at runtime. This can be any folder
+containing `.wasm` files.
+
+During local development,
+when a short feedback loop is desired, you can set this parameter to
+your build toolchain's output directory (wherever the compiled
+`<filter-name>.wasm` file is produced):
+
+```console
+$ export KONG_WASM_FILTERS_PATH=/path/to/my_filter/build
+```
+
+#### Link to a Kong service and route
+
+Now, the next step is to create a Kong filter chain entity using the
```

Comment:
In the Gateway documentation, all entities part of the DB model are capitalized; it should also be the case for "Filter Chains".